### ISSUE TITLE
Incorrect types of the data.kubernetes_service attributes

### ISSUE THREAD

**Issue by minherz** [hashicorp/terraform-provider-kubernetes#2038]
The official K8s [documentation][doc1] describes the status as a _structure_ containing (among other fields) an _object_ of `loadBalancer`. However, the provider defines the schema for the `kubernetes_service` data source with both `status` and `load_balancer` as lists of objects:

https://github.com/hashicorp/terraform-provider-kubernetes/blob/fab1450d769843495c370f2431a8192aaa5e5321/kubernetes/data_source_kubernetes_service.go#L207-L218

The [same schema][doc2] is implemented for the definition of the `kubernetes_service` resource attributes. 
It is confusing and contra-intuitive. It also mismatch the K8s object model.

[doc1]: https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceStatus
[doc2]: https://github.com/hashicorp/terraform-provider-kubernetes/blob/f0d0f536b7c6e38c0bce71c590c0ae3d841e94fa/kubernetes/resource_kubernetes_service.go#L299-L307

**Comment by minherz** [comment]
My apologizes, indeed I skipped the most relevant part of the problem description. I've updated the problem in the description. Also copy/pasting it here:

When accessing `kubernetes_service` attribute `status` to retrieve the load balancer IP need to write something like the following: `kubernetes_service.my_service.status[0].load_balancer[0].ingress[0].ip`. And while it is true that the `core/v1` schema defines `LoadBalancerIngress` as an array, it is not true for the `ServiceStatus` and the `LoadBalancerStatus`. However, the implementation in the provider [does define][1] it this way.

[1]: https://github.com/hashicorp/terraform-provider-kubernetes/blob/f0d0f536b7c6e38c0bce71c590c0ae3d841e94fa/kubernetes/resource_kubernetes_service.go#L299-L307
